Extension/src/LanguageServer/copilotCompletionContextProvider.ts

```diff
@@ -123,8 +129,19 @@ export class CopilotCompletionContextProvider implements ContextResolver<CodeSni
             const client = clients.getClientFor(docUri);
             if (!client) { throw WellKnownErrors.clientNotFound(); }
             const getCompletionContextStartTime = performance.now();
+
+            // Start collection of project traits concurrently with the completion context computation.
+            const telemetryProperties: Record<string, string> = {};
+            const telemetryMetrics: Record<string, number> = {};
+            const projectContextPromise = (async (): Promise<ProjectContext | undefined> => {
+                const getProjectContextStartTime = performance.now();
+                const projectContext = await getProjectContext(docUri, { flags: {} }, telemetryProperties, telemetryMetrics);

```

Comment:
Should this be cached? It seems strange to re-request this on every completion. Wasn't the previous related files version cached for like 2 minutes or something like that?

Reply:
Yes, it does. I'll create a note to add it in a subsequent PR (or anyway asap)